lib/util/ssr.js

```diff
@@ -6,13 +6,45 @@
  */
 'use strict';
 
+let ssrEscapeFound = false;
+
 module.exports.isSSREscape = function isSSREscape(node) {
-    return (
-        (node.type === 'IfStatement' || node.type === 'ConditionalExpression') &&
-        (isMetaEnvCheck(node.test) || isWindowOrDocumentCheck(node.test))
-    );
+    if (node.type === 'IfStatement' || node.type === 'ConditionalExpression') {
+        ssrEscapeFound = false;
+        if (checkConditionalStatements(node.test) || isWindowOrDocumentCheck(node.test)) {
+            return true;
+        }
+    }
+
+    return false;
 };
 
+function checkConditionalStatements(test) {
+    let node = test;
+
+    // Recursive Case: If the node is a logical expression, check its left and right parts.
+    if (node.type === 'LogicalExpression' || node.type === 'BinaryExpression') {
+        checkConditionalStatements(node.left);
+        checkConditionalStatements(node.right);
+    }
+
+    // Base Case: If the node is an Identifier or MemberExpression, don't need to continue.
```

Comment:
Why?

Reply:
This would be a helpful comment! 🙂

Reply:
Actually, this `if` seems redundant to me since we fall through to `return false` on line 47 anyway. I think we can remove it entirely.